### PR TITLE
Provide a log filepath that complies with our requirements (CRAFT-96).

### PR DESCRIPTION
**Pull request by facundobatista**
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

### CODE REVIEW

File: craft_cli/messages.py [hunk 1]
```diff
@@ -39,23 +41,51 @@ class _MessageInfo:
...
     return shutil.get_terminal_size().columns
 
 
-def get_log_filepath(appname):
-    """Provide a filepath for logging into."""
-    # XXX Facundo 2021-09-03: this will change heavily in next couple of branches
-    _, filepath = tempfile.mkstemp(prefix=f"{appname}-")
-    return filepath
+def get_log_filepath(appname: str) -> pathlib.Path:
+    """Provide a filepath for logging into.
```
Comment (sergiusens) [inline craft_cli/messages.py:54]:
```suggestion
    """Provide a unique filepath for logging.
```

File: craft_cli/messages.py [hunk 2]
```diff
@@ -39,23 +41,51 @@ class _MessageInfo:
 # the different modes the Emitter can be set
 EmitterMode = enum.Enum("EmitterMode", "QUIET NORMAL VERBOSE TRACE")
 
+# the limit to how many log files to have
+LOG_FILES_LIMIT = 5
+
```
Comment (sergiusens) [inline craft_cli/messages.py:46]: This renders nicely in sphinx

```suggestion
LOG_FILES_LIMIT = 5
"""The limit to how many log files to have."""
```

I might personally prefer MAX over LIMIT

File: craft_cli/messages.py [hunk 3]
```diff
@@ -39,23 +41,51 @@ class _MessageInfo:
...
-    return filepath
+def get_log_filepath(appname: str) -> pathlib.Path:
+    """Provide a filepath for logging into.
+
+    The app name is used for both the directory where the logs are located and each log name.
+
+    Rules:
+    - use an appdirs provided directory
+    - base filename is <appname>.<timestamp with microseconds>.log
+    - it rotates until it gets to ROTATION_LIMIT
```
Comment (sergiusens) [inline craft_cli/messages.py:61]: ROTATION_LIMIT is probably the other constant
```suggestion
    - it rotates until it reaches :data:`.LOG_FILES_LIMIT`
```

File: craft_cli/messages.py [hunk 4]
```diff
@@ -39,23 +41,51 @@ class _MessageInfo:
...
+    - after limit is achieved, remove the exceeding files
+    - ignore other non-log files in the directory
+
+    Existing files are not renamed (no need, as each name is unique) nor gzipped (they may
+    be currently in use by other process).
```
Comment (sergiusens) [inline craft_cli/messages.py:66]:
```suggestion
    be currently in use by another process).
```

File: craft_cli/messages.py [hunk 5]
```diff
@@ -39,23 +41,51 @@ class _MessageInfo:
...
+    """
+    basedir = pathlib.Path(appdirs.user_log_dir()) / appname
+    filename = f"{appname}-{datetime.now():%Y%m%d-%H%M%S.%f}.log"
+
+    # ensure the basedir is there
+    basedir.mkdir(exist_ok=True)
+
+    # check if we doesn't have too many logs in the dir, and remove the exceeding ones (note
```
Comment (sergiusens) [inline craft_cli/messages.py:74]:
```suggestion
    # check if we have too many logs in the dir, and remove the exceeding ones (note
```

File: craft_cli/messages.py (hunk 2 above)
Comment (cmatsuoka) [reply craft_cli/messages.py:46]: Agree with Sergio, `MAX_LOG_FILES` could be better here.

File: craft_cli/messages.py (hunk 1 above)
Comment (facundobatista) [reply craft_cli/messages.py:54]: Improved.

File: craft_cli/messages.py (hunk 2 above)
Comment (facundobatista) [reply craft_cli/messages.py:46]: Changed to `_MAX_LOG_FILES`, to also make it private (not intended to be part of the public API)

File: craft_cli/messages.py (hunk 3 above)
Comment (facundobatista) [reply craft_cli/messages.py:61]: Improved. Don't know how to really see this, though.

Comment (sergiusens) [reply craft_cli/messages.py]: "make docs" at the root of the project should construct documentation

**Review by cmatsuoka — APPROVED** [review]
Thanks Facundo, lgtm.

File: craft_cli/messages.py (hunk 3 above)
Comment (facundobatista) [reply craft_cli/messages.py]: It did, thanks.

File: craft_cli/messages.py [hunk 6]
```diff
@@ -51,30 +51,30 @@ def get_terminal_width() -> int:
...
-    """Provide a filepath for logging into.
+    """Provide a unique filepath for logging.
 
     The app name is used for both the directory where the logs are located and each log name.
 
     Rules:
     - use an appdirs provided directory
     - base filename is <appname>.<timestamp with microseconds>.log
-    - it rotates until it gets to ROTATION_LIMIT
+    - it rotates until it gets to reaches :data:`._MAX_LOG_FILES`
```
Comment (sergiusens) [inline craft_cli/messages.py:61]: I would make this public as it is documented

File: craft_cli/messages.py [hunk 7]
```diff
@@ -42,7 +42,7 @@ class _MessageInfo:
 EmitterMode = enum.Enum("EmitterMode", "QUIET NORMAL VERBOSE TRACE")
 
 # the limit to how many log files to have
-LOG_FILES_LIMIT = 5
+_MAX_LOG_FILES = 5
```
Comment (sergiusens) [inline craft_cli/messages.py:45]: I would make this public again and document it so a user of the API is exposed to the default in documentation.

**Review by sergiusens — APPROVED** [review]
Looks good, I would prefer for the constant to be public as it is documented.

File: craft_cli/messages.py (hunk 6 above)
Comment (facundobatista) [reply craft_cli/messages.py:61]: As we talked, made everything private.

File: craft_cli/messages.py (hunk 7 above)
Comment (facundobatista) [reply craft_cli/messages.py:45]: As we talked, made everything private.

**Pull request merged**